src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2881,6 +2888,176 @@ public ConstructorEdit(INamedTypeSymbol oldType)
+                var oldType = oldSymbol as INamedTypeSymbol;
+                if (newType.DelegateInvokeMethod is not null)
+                {
+                    // If this is a delegate with attributes on its return type for example, they are found on the DelegateInvokeMethod
+                    ReportAttributeEdits(oldType?.DelegateInvokeMethod, newType.DelegateInvokeMethod, capabilities, diagnostics, semanticEdits, syntaxMap, cancellationToken);
+                }
+            }
+
+            foreach (var parameter in newSymbol.GetParameters())
+            {
+                var oldParameter = oldSymbol?.GetParameters().SingleOrDefault(p => p.Name.Equals(parameter.Name));
+                ReportAttributeRudeEdits(oldParameter?.GetAttributes(), parameter.GetAttributes(), parameter, capabilities, diagnostics, out attributesHaveChanged);
+                needsEdit |= attributesHaveChanged;
+            }
+
+            foreach (var typeParam in newSymbol.GetTypeParameters())
+            {
+                var oldParameter = oldSymbol?.GetTypeParameters().SingleOrDefault(p => p.Name.Equals(typeParam.Name));
+                ReportAttributeRudeEdits(oldParameter?.GetAttributes(), typeParam.GetAttributes(), typeParam, capabilities, diagnostics, out attributesHaveChanged);
+                needsEdit |= attributesHaveChanged;
+            }
+
+            // This is the only case we care about whether to issue an edit or not, because this is the only case where types have their attributes checked
+            // and types are the only things that would otherwise not have edits reported.
+            ReportAttributeRudeEdits(oldSymbol?.GetAttributes(), newSymbol.GetAttributes(), newSymbol, capabilities, diagnostics, out attributesHaveChanged);
+            needsEdit |= attributesHaveChanged;
+
+            // If we don't need to add an edit, then we're done
+            if (!needsEdit || semanticEdits is null)
+            {
+                return;
+            }
+
+            // Most symbol types will automatically have an edit added, so we just need to handle a few
+            if (newSymbol is INamedTypeSymbol)
+            {
+                var symbolKey = SymbolKey.Create(newSymbol, cancellationToken);
+                semanticEdits.Add(new SemanticEditInfo(SemanticEditKind.Update, symbolKey, syntaxMap, null, null));
+            }
+            else if (newSymbol is IMethodSymbol { MethodKind: MethodKind.DelegateInvoke })
+            {
+                var symbolKey = SymbolKey.Create(newSymbol.ContainingSymbol, cancellationToken);
+                semanticEdits.Add(new SemanticEditInfo(SemanticEditKind.Update, symbolKey, syntaxMap, null, null));
+            }
+        }
+
+        private void ReportAttributeRudeEdits(ImmutableArray<AttributeData>? oldAttributes, ImmutableArray<AttributeData> newAttributes, ISymbol newSymbol, EditAndContinueCapabilities capabilities, ArrayBuilder<RudeEditDiagnostic> diagnostics, out bool needsEdit)
```

Comment:
>ReportAttributeRudeEdits [](http://example.com/codeflow?start=21&length=24)

`bool HasCustomAttributeChanges`?

Reply:
I could refactor it to be that I guess, and move the diagnostics adding out to the calling method, but its called a few times. Will see.

Reply:
Left the diagnostics, but made it still return bool. Pulling the diagnostic out was a bit messy because of the property accesser handling